### PR TITLE
Fix/rx bloc cli template bug fixing

### PR DESCRIPTION
**Pull request by PVGeorgievPrime**

### CODE REVIEW

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.g.dart [hunk 1]
```diff
@@ -63,6 +63,28 @@ class _AuthDataSource implements AuthDataSource {
     return null;
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.g.dart:63]: Please do not commit generated files. They are now generated during as part of the `rx_bloc_cli create` command
Screenshot 2023-03-06 at 11 01 20

File: packages/rx_bloc_cli/lib/src/rx_bloc_cli_constants.dart [hunk 2]
```diff
@@ -1,2 +1,2 @@
 /// Indicates the version of the package
-const rxBlocCliPackageVersion = '1.4.1';
+const rxBlocCliPackageVersion = '2.0.0';
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/lib/src/rx_bloc_cli_constants.dart:2]: Good catch!

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/bin/server/controllers/authentication_controller.dart [hunk 3]
```diff
@@ -31,6 +31,11 @@ class AuthenticationController extends ApiController {
       '/api/logout',
       _logoutHandler,
     );
+    router.addRequest(
+      RequestType.POST,
+      '/api/auth/validate-token',
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/bin/server/controllers/authentication_controller.dart:36]: In the OAuth2 specification there is no such think as `validate-token`

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.dart [hunk 4]
```diff
@@ -17,4 +17,7 @@ abstract class AuthDataSource {
 
   @POST('/api/logout')
   Future<void> logout();
+
+  @POST('/api/auth/validate-token')
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.dart:21]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.dart [hunk 5]
```diff
@@ -6,11 +6,17 @@ part 'authenticate_user_request_model.g.dart';
...
+    this.username,
+    this.password,
+    this.refreshToken,
+    this.token,
+  });
 
   final String? username;
   final String? password;
   final String? refreshToken;
+  final String? token;
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.dart:19]: It looks like that you have mixed the authentication process with the one that retrieves a new access token. Please create separate models for those two flows

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.g.dart [hunk 6]
```diff
@@ -7,11 +7,12 @@ part of 'authenticate_user_request_model.dart';
 // **************************************************************************
 
 AuthUserRequestModel _$AuthUserRequestModelFromJson(
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.g.dart:9]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart [hunk 7]
```diff
@@ -59,4 +59,10 @@ class AuthRepository {
       );
 
   Future<void> logout() => _errorMapper.execute(() => _authDataSource.logout());
+
+  Future<void> validateAccessToken(String token) => _errorMapper.execute(
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart:63]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart [hunk 8]
```diff
@@ -43,13 +43,13 @@ class AuthRepository {
   /// Fetch a new access token using the current refresh token
   Future<AuthTokenModel> fetchNewToken(String refreshToken) =>
       _errorMapper.execute(() => _refreshTokenDataSource.refresh(
-        AuthUserRequestModel(refreshToken: refreshToken),
-      ));
+            AuthUserRequestModel(refreshToken: refreshToken),
+          ));
 
   Future<AuthTokenModel> authenticate(
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart:49]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart [hunk 9]
```diff
@@ -14,7 +14,16 @@ class AuthService {
   /// Checks if the user is authenticated.
   Future<bool> isAuthenticated() async {
     final token = await getToken();
-    return token != null;
+    if (token == null) {
+      return false;
+    }
+
+    try {
+      await _authRepository.validateAccessToken(token);
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart:22]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart [hunk 10]
```diff
@@ -48,7 +57,7 @@ class AuthService {
   /// Checks the user credentials passed in [email], [password] and returns
   /// auth token.
   Future<AuthTokenModel> authenticate(
-      {String? email, String? password, String? refreshToken}) =>
+          {String? email, String? password, String? refreshToken}) =>
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart:60]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.dart [hunk 11]
```diff
@@ -1,4 +1,4 @@
-{{> licence.dart }}
+// {{> licence.dart }}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.dart:1]: Please uncomment the license snippet

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.rxb.g.dart [hunk 12]
```diff
@@ -29,7 +29,7 @@ abstract class $RouterBloc extends RxBlocBase
   final _$pushEvent = PublishSubject<_PushEventArgs>();
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.rxb.g.dart:29]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/base/common_blocs/router_bloc_mock.mocks.dart [hunk 13]
```diff
@@ -1,13 +1,11 @@
 // Mocks generated by Mockito 5.3.2 from annotations
-// in test_app/test/base/common_blocs/router_bloc_mock.dart.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/base/common_blocs/router_bloc_mock.mocks.dart:2]: Same as above

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/bin/server/controllers/authentication_controller.dart (hunk 3 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/bin/server/controllers/authentication_controller.dart]: removed

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.dart (hunk 4 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/data_sources/remote/auth_data_source.dart]: removed

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.dart (hunk 5 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.dart]: removed

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart (hunk 9 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/services/auth_service.dart]: removed

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.dart (hunk 11 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.dart]: done

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.rxb.g.dart (hunk 12 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_router/blocs/router_bloc.rxb.g.dart]: done

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.g.dart (hunk 6 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/models/request_models/authenticate_user_request_model.g.dart]: done

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart (hunk 7 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_auth/repositories/auth_repository.dart]: removed

File: packages/rx_bloc_cli/CHANGELOG.md [hunk 14]
```diff
@@ -1,4 +1,7 @@
-## [2.0.0] - TBD
+## [2.1.0] - TBD
+* Bug fixing
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/CHANGELOG.md:2]: Add information for the dashboard feature and the new param --enable-feature-counter

**Review by StanevPrime — APPROVED** [review]

File: packages/rx_bloc_cli/CHANGELOG.md (hunk 14 above)
Comment (PVGeorgievPrime) [reply packages/rx_bloc_cli/CHANGELOG.md:4]: Done

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**